atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,467 @@
+import java.io.IOException;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class OracleNamespaceDeleterTests {
+    @ClassRule
+    public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+    private static final String TABLE_PREFIX = "a_";
+    private static final String OVERFLOW_TABLE_PREFIX = "ao_";
+    private static final String ANOTHER_TABLE_PREFIX = "test_";
+    private static final String ANOTHER_OVERFLOW_TABLE_PREFIX = "over_";
+    private static final String TABLE_NAME_1 = "abc";
+    private static final String TABLE_NAME_2 = "hello";
+    private static final String TABLE_NAME_3 = "world";
+    private static final String TEST_USER = "testuser";
+    private static final String TEST_USER_2 = "testuser2";
+
+    private ExecutorService executorService;
+    private SqliteOracleAdapter sqliteOracleAdapter;
+
+    @Before
+    public void before() throws IOException {
+        executorService = MoreExecutors.newDirectExecutorService();
+        sqliteOracleAdapter = new SqliteOracleAdapter(TEMPORARY_FOLDER.newFile());
+        sqliteOracleAdapter.initializeMetadataAndMappingTables();
+    }
+
+    @After
+    public void after() {
+        executorService.shutdown();
+        sqliteOracleAdapter.close();
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceOnlyDropsTablesWithConfigPrefixesForCurrentUser() {
+        NamespaceDeleter namespaceDeleter = createDefaultNamespaceDeleter();
+        Set<TableDetails> tablesToDelete = Sets.union(
+                createTablesWithDefaultPrefixes(TABLE_NAME_1), createTablesWithDefaultPrefixes(TABLE_NAME_2));
+
+        Set<TableDetails> tablesThatShouldNotBeDeleted = Set.of(
+                createTable(ANOTHER_TABLE_PREFIX, TABLE_NAME_1, TEST_USER),
+                createTable(ANOTHER_OVERFLOW_TABLE_PREFIX, TABLE_NAME_2, TEST_USER),
+                createTable(TABLE_PREFIX, TABLE_NAME_3, TEST_USER_2),
+                // In most SQL implementations, _ is the single character wildcard (so a_ matches ab too).
+                // We obviously do not want to drop a table named abc when the prefix is a_, so the following explicitly
+                // tests that case.
+                createTable("ab_", TABLE_NAME_1, TEST_USER));
+
+        assertThatTableDetailsMatchPersistedData(Sets.union(tablesToDelete, tablesThatShouldNotBeDeleted));
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertThatTableDetailsMatchPersistedData(tablesThatShouldNotBeDeleted);
+    }
+
+    @Test
```

Comment:
Maybe silly: I'd probably just add one with no tables at all, just to make sure it works